include/firestarter/Environment/X86/Payload/AVX512_AMX_Payload.hpp

```diff
@@ -0,0 +1,58 @@
+/******************************************************************************
+ * FIRESTARTER - A Processor Stress Test Utility
+ * Copyright (C) 2020 TU Dresden, Center for Information Services and High
+ * Performance Computing
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/\>.
+ *
+ * Contact: daniel.hackenberg@tu-dresden.de
+ *****************************************************************************/
+
+#pragma once
+
+#include <firestarter/Environment/X86/Payload/X86Payload.hpp>
+
+namespace firestarter::environment::x86::payload {
+class AVX512_AMX_Payload final : public X86Payload {
+public:
+  AVX512_AMX_Payload(asmjit::CpuFeatures const &supportedFeatures)
+      : X86Payload(supportedFeatures, {asmjit::CpuFeatures::X86::kAMX_BF16},
```

Comment:
Checking for the kAMX_BF16 feature may have prevented you from integrating the AMX code it into the AVX512 payload. Can we check this feature if the "AMX" meta-instruction is selection during the compilePayload step instead?